### PR TITLE
Improve building

### PR DESCRIPTION
**Pull request by sunapi386**
Here are some changes needed for making on Ubuntu 18.04.

### CODE REVIEW

File: sendrecv/gst/Makefile [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
-LIBS	:= $(shell pkg-config --libs --cflags gstreamer-webrtc-1.0 gstreamer-sdp-1.0 libsoup-2.4 json-glib-1.0)
-CFLAGS	:= -O0 -ggdb -Wall -fno-omit-frame-pointer
+CC     := gcc
+LIBS   := $(shell pkg-config --libs --cflags gstreamer-webrtc-1.0 gstreamer-sdp-1.0 libsoup-2.4 json-glib-1.0)
+CFLAGS := -O0 -ggdb -Wall -fno-omit-frame-pointer \
+          -I/usr/include/gstreamer-1.0 \
+          -I/usr/include/glib-2.0 \
+          -I/usr/lib/x86_64-linux-gnu/glib-2.0/include \
+          -I/usr/include/libsoup-2.4 \
+          -I/usr/include/json-glib-1.0
```
Comment (ystreet) [inline sendrecv/gst/Makefile]: explicit paths are a no go.  These paths should be constructed from `pkg-config --cflags` in a similar way to the original `LIBS` above

File: README.md [hunk 2]
```diff
@@ -18,6 +18,10 @@ One thing to note is that it's written in Python 2, so you may need to replace a
 
 ## Building GStreamer manually from source
 
+```
+sudo apt-get install -y gstreamer1.0-tools gstreamer1.0-nice gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly gstreamer1.0-plugins-good libgstreamer1.0-dev git libglib2.0-dev libgstreamer-plugins-bad1.0-dev libsoup2.4-dev libjson-glib-dev
```
Comment (ystreet) [inline README.md:24]: This should be prefixed with the distribution these commands are for. Ubuntu/debian/etc in this case.

**Review by ystreet — CHANGES_REQUESTED** [review]

File: sendrecv/gst/Makefile (hunk 1 above)
Comment (sunapi386) [reply sendrecv/gst/Makefile]: Fixed.

File: README.md (hunk 2 above)
Comment (sunapi386) [reply README.md:24]: It sounds like you want me to clarify which distro in a comment, so I added a comment.

**Review by sunapi386 — COMMENTED** [review]
Addressed comments.

**Comment by sunapi386** [comment]
@ystreet PTAL.

File: sendrecv/gst/Makefile [hunk 3]
```diff
@@ -1,6 +1,6 @@
-CC	:= gcc
-LIBS	:= $(shell pkg-config --libs --cflags gstreamer-webrtc-1.0 gstreamer-sdp-1.0 libsoup-2.4 json-glib-1.0)
-CFLAGS	:= -O0 -ggdb -Wall -fno-omit-frame-pointer
-
+CC     := gcc
+LIBS   := $(shell pkg-config --libs --cflags gstreamer-webrtc-1.0 gstreamer-sdp-1.0 libsoup-2.4 json-glib-1.0)
+CFLAGS := -O0 -ggdb -Wall -fno-omit-frame-pointer \
+          $(shell pkg-config --cflags gstreamer-1.0 glib-2.0 libsoup-2.4 json-glib-1.0 glib-2.0)
```
Comment (ystreet) [inline sendrecv/gst/Makefile]: The packages in the cflags list need to match the ones in the `LIBS` line.  i.e. `gstreamer-webrtc-1.0` and `gstreamer-sdp-1.0` are missing

**Review by ystreet — CHANGES_REQUESTED** [review]

File: sendrecv/gst/Makefile (hunk 3 above)
Comment (sunapi386) [reply sendrecv/gst/Makefile]: Oh ok, I added this. I guess `"$(CC)" $(CFLAGS) $^ $(LIBS) -o $@` needs the flags to match the libs. Do I need to make them match 1:1?

Comment (ystreet) [reply sendrecv/gst/Makefile]: They should match 1:1 yes.

**Comment by ystreet** [comment]
@sunapi386 Are you still planning to look at this?

File: sendrecv/gst/Makefile (hunk 3 above)
Comment (sunapi386) [reply sendrecv/gst/Makefile]: I made them match
```
LIBS   := $(shell pkg-config --libs --cflags glib-2.0 gstreamer-1.0 gstreamer-sdp-1.0 gstreamer-webrtc-1.0 json-glib-1.0 libsoup-2.4)
CFLAGS := -O0 -ggdb -Wall -fno-omit-frame-pointer \
		$(shell pkg-config --cflags glib-2.0 gstreamer-1.0 gstreamer-sdp-1.0 gstreamer-webrtc-1.0 json-glib-1.0 libsoup-2.4)
```

**Comment by sunapi386** [comment]
@ystreet looked into it, I think I resolved it.

**Comment by sunapi386** [comment]
@ystreet PTAL and let me know.

**Review by ystreet — APPROVED** [review]

**Comment by ystreet** [comment]
Thanks for working on this!

**Pull request merged**